js/ui/notificationDaemon.js

```diff
@@ -762,6 +764,10 @@ const GtkNotificationDaemonAppSource = new Lang.Class({
         }
         return [this._appId, notifications];
     },
+
+    get app() {
+        return this._app;
+    }
```

Comment:
Ah, got it, this is used in

    if (source.app != this.app)
              return;